lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_emm_encode_decode.cpp

```diff
@@ -15,21 +15,21 @@
 extern "C" {
 #include "lte/gateway/c/core/common/dynamic_memory_check.h"
 #include "lte/gateway/c/core/oai/common/log.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AttachAccept.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AttachRequest.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AuthenticationReject.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AuthenticationRequest.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AuthenticationResponse.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/CsServiceNotification.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/EmmInformation.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/ExtendedServiceRequest.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeCommand.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeComplete.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/ServiceRequest.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateAccept.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateReject.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateRequest.h"
-#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/UplinkNasTransport.h"
+#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AttachAccept.hpp"
```

Comment:
These files need not be under extern "C"

Reply:
That's right. This extern "C" was causing undefined reference to encode_attach_accept/decode_attach_accept functions in AttachAccept.hpp due to which they had to be put under            
 #ifdef __cplusplus
extern "C" {
#endif
Thanks @rsarwad

Reply:
Shall we discuss because I branched out this code, compiled and executed unit test case locally.